### DISCUSSION TITLE
Upgrading to 0.17 using deb packages - error

### DISCUSSION THREAD

**Discussion by tomd-tc** [inventree/InvenTree#8737] [General]
Congratulations on the 0.17 release. It has been a much awaited release...

I've been trying to upgrade it using the DEB packages, however, during the installation and then trying to run I get the error message below. I've tried reinstalling the PIL library as suggested but to no avail. Any suggestions on what next steps to take?
 
 (env) root@wms:/opt/inventree# inventree run invoke update

#Updating InvenTree installation...
Installing required python packages from '/opt/inventree/src/backend/requirements.txt'
WARNING: Ignoring invalid distribution ~etuptools (/opt/inventree/env/lib/python3.11/site-packages)
Requirement already satisfied: pip in ./env/lib/python3.11/site-packages (24.3.1)
Requirement already satisfied: setuptools in ./env/lib/python3.11/site-packages (75.6.0)
WARNING: Ignoring invalid distribution ~etuptools (/opt/inventree/env/lib/python3.11/site-packages)
WARNING: Ignoring invalid distribution ~etuptools (/opt/inventree/env/lib/python3.11/site-packages)
WARNING: Ignoring invalid distribution ~etuptools (/opt/inventree/env/lib/python3.11/site-packages)
Requirement already satisfied: asgiref==3.8.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 3)) (3.8.1)
Requirement already satisfied: async-timeout==5.0.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 11)) (5.0.1)
Requirement already satisfied: attrs==24.2.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 15)) (24.2.0)
Requirement already satisfied: babel==2.16.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 21)) (2.16.0)
Requirement already satisfied: bleach==6.2.0 in ./env/lib/python3.11/site-packages (from bleach[css]==6.2.0->-r /opt/inventree/src/backend/requirements.txt (line 25)) (6.2.0)
Requirement already satisfied: brotli==1.1.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 29)) (1.1.0)
Requirement already satisfied: certifi==2024.8.30 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 156)) (2024.8.30)
Requirement already satisfied: cffi==1.17.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 162)) (1.17.1)
Requirement already satisfied: charset-normalizer==3.4.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 233)) (3.4.0)
Requirement already satisfied: coreapi==2.3.3 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 340)) (2.3.3)
Requirement already satisfied: coreschema==0.0.4 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 344)) (0.0.4)
Requirement already satisfied: cryptography==43.0.3 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 348)) (43.0.3)
Requirement already satisfied: cssselect2==0.7.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 379)) (0.7.0)
Requirement already satisfied: defusedxml==0.7.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 383)) (0.7.1)
Requirement already satisfied: deprecated==1.2.15 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 389)) (1.2.15)
Requirement already satisfied: diff-match-patch==20241021 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 397)) (20241021)
Requirement already satisfied: dj-rest-auth==7.0.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 401)) (7.0.0)
Requirement already satisfied: django==4.2.17 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 404)) (4.2.17)
Requirement already satisfied: django-allauth==65.2.0 in ./env/lib/python3.11/site-packages (from django-allauth[openid,saml]==65.2.0->-r /opt/inventree/src/backend/requirements.txt (line 440)) (65.2.0)
Requirement already satisfied: django-allauth-2fa==0.11.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 445)) (0.11.1)
Requirement already satisfied: django-cleanup==9.0.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 449)) (9.0.0)
Requirement already satisfied: django-cors-headers==4.6.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 453)) (4.6.0)
Requirement already satisfied: django-crispy-forms==1.14.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 457)) (1.14.0)
Requirement already satisfied: django-dbbackup==4.2.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 461)) (4.2.1)
Requirement already satisfied: django-error-report-2==0.4.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 465)) (0.4.2)
Requirement already satisfied: django-filter==24.3 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 469)) (24.3)
Requirement already satisfied: django-flags==5.0.13 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 473)) (5.0.13)
Requirement already satisfied: django-formtools==2.5.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 477)) (2.5.1)
Requirement already satisfied: django-ical==1.9.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 481)) (1.9.2)
Requirement already satisfied: django-import-export==3.3.9 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 485)) (3.3.9)
Requirement already satisfied: django-ipware==7.0.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 489)) (7.0.1)
Requirement already satisfied: django-js-asset==2.2.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 493)) (2.2.0)
Requirement already satisfied: django-maintenance-mode==0.21.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 497)) (0.21.1)
Requirement already satisfied: django-markdownify==0.9.5 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 501)) (0.9.5)
Requirement already satisfied: django-money==3.2.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 505)) (3.2.0)
Requirement already satisfied: django-mptt==0.16.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 509)) (0.16.0)
Requirement already satisfied: django-otp==1.5.4 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 513)) (1.5.4)
Requirement already satisfied: django-picklefield==3.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 517)) (3.2)
Requirement already satisfied: django-q-sentry==0.1.6 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 521)) (0.1.6)
Requirement already satisfied: django-q2==1.7.4 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 524)) (1.7.4)
Requirement already satisfied: django-recurrence==1.11.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 528)) (1.11.1)
Requirement already satisfied: django-redis==5.4.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 532)) (5.4.0)
Requirement already satisfied: django-sesame==3.2.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 536)) (3.2.2)
Requirement already satisfied: django-sql-utils==0.7.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 540)) (0.7.0)
Requirement already satisfied: django-sslserver==0.22 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 544)) (0.22)
Requirement already satisfied: django-stdimage==6.0.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 547)) (6.0.2)
Requirement already satisfied: django-structlog==8.1.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 551)) (8.1.0)
Requirement already satisfied: django-taggit==6.1.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 555)) (6.1.0)
Requirement already satisfied: django-user-sessions==2.0.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 559)) (2.0.0)
Requirement already satisfied: django-weasyprint==2.3.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 563)) (2.3.0)
Requirement already satisfied: django-xforwardedfor-middleware==2.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 567)) (2.0)
Requirement already satisfied: djangorestframework==3.14.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 570)) (3.14.0)
Requirement already satisfied: djangorestframework-simplejwt==5.3.1 in ./env/lib/python3.11/site-packages (from djangorestframework-simplejwt[crypto]==5.3.1->-r /opt/inventree/src/backend/requirements.txt (line 578)) (5.3.1)
Requirement already satisfied: drf-spectacular==0.27.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 582)) (0.27.2)
Requirement already satisfied: dulwich==0.22.6 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 586)) (0.22.6)
Requirement already satisfied: et-xmlfile==2.0.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 630)) (2.0.0)
Requirement already satisfied: feedparser==6.0.11 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 634)) (6.0.11)
Requirement already satisfied: flexcache==0.3 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 638)) (0.3)
Requirement already satisfied: flexparser==0.4 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 642)) (0.4)
Requirement already satisfied: fonttools==4.55.0 in ./env/lib/python3.11/site-packages (from fonttools[woff]==4.55.0->-r /opt/inventree/src/backend/requirements.txt (line 646)) (4.55.0)
Requirement already satisfied: googleapis-common-protos==1.66.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 698)) (1.66.0)
Requirement already satisfied: grpcio==1.68.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 704)) (1.68.0)
Requirement already satisfied: gunicorn==23.0.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 763)) (23.0.0)
Requirement already satisfied: html5lib==1.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 767)) (1.1)
Requirement already satisfied: icalendar==6.1.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 771)) (6.1.0)
Requirement already satisfied: idna==3.10 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 775)) (3.10)
Requirement already satisfied: importlib-metadata==8.5.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 779)) (8.5.0)
Requirement already satisfied: inflection==0.5.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 786)) (0.5.1)
Requirement already satisfied: isodate==0.7.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 790)) (0.7.2)
Requirement already satisfied: itypes==1.2.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 794)) (1.2.0)
Requirement already satisfied: jinja2==3.1.4 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 798)) (3.1.4)
Requirement already satisfied: jsonschema==4.23.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 802)) (4.23.0)
Requirement already satisfied: jsonschema-specifications==2024.10.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 806)) (2024.10.1)
Requirement already satisfied: lxml==5.3.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 810)) (5.3.0)
Requirement already satisfied: markdown==3.7 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 952)) (3.7)
Requirement already satisfied: markuppy==1.14 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 956)) (1.14)
Requirement already satisfied: markupsafe==3.0.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 959)) (3.0.2)
Requirement already satisfied: odfpy==1.4.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1022)) (1.4.1)
Requirement already satisfied: openpyxl==3.1.5 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1025)) (3.1.5)
Requirement already satisfied: opentelemetry-api==1.28.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1029)) (1.28.2)
Requirement already satisfied: opentelemetry-exporter-otlp==1.28.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1043)) (1.28.2)
Requirement already satisfied: opentelemetry-exporter-otlp-proto-common==1.28.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1047)) (1.28.2)
Requirement already satisfied: opentelemetry-exporter-otlp-proto-grpc==1.28.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1053)) (1.28.2)
Requirement already satisfied: opentelemetry-exporter-otlp-proto-http==1.28.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1057)) (1.28.2)
Requirement already satisfied: opentelemetry-instrumentation==0.49b2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1061)) (0.49b2)
Requirement already satisfied: opentelemetry-instrumentation-django==0.49b2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1069)) (0.49b2)
Requirement already satisfied: opentelemetry-instrumentation-redis==0.49b2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1073)) (0.49b2)
Requirement already satisfied: opentelemetry-instrumentation-requests==0.49b2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1077)) (0.49b2)
Requirement already satisfied: opentelemetry-instrumentation-wsgi==0.49b2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1081)) (0.49b2)
Requirement already satisfied: opentelemetry-proto==1.28.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1085)) (1.28.2)
Requirement already satisfied: opentelemetry-sdk==1.28.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1092)) (1.28.2)
Requirement already satisfied: opentelemetry-semantic-conventions==0.49b2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1099)) (0.49b2)
Requirement already satisfied: opentelemetry-util-http==0.49b2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1109)) (0.49b2)
Requirement already satisfied: packaging==24.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1116)) (24.2)
Requirement already satisfied: pdf2image==1.17.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1122)) (1.17.0)
Requirement already satisfied: pillow==11.0.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1126)) (11.0.0)
Requirement already satisfied: pint==0.24.4 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1209)) (0.24.4)
Requirement already satisfied: pip-licenses==5.0.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1213)) (5.0.0)
Requirement already satisfied: platformdirs==4.3.6 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1217)) (4.3.6)
Requirement already satisfied: prettytable==3.12.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1221)) (3.12.0)
Requirement already satisfied: protobuf==5.28.3 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1225)) (5.28.3)
Requirement already satisfied: py-moneyed==3.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1240)) (3.0)
Requirement already satisfied: pycparser==2.22 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1244)) (2.22)
Requirement already satisfied: pydyf==0.10.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1248)) (0.10.0)
Requirement already satisfied: pyjwt==2.10.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1254)) (2.10.1)
Requirement already satisfied: pyphen==0.17.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1258)) (0.17.0)
Requirement already satisfied: python-barcode==0.15.1 in ./env/lib/python3.11/site-packages (from python-barcode[images]==0.15.1->-r /opt/inventree/src/backend/requirements.txt (line 1262)) (0.15.1)
Requirement already satisfied: python-dateutil==2.9.0.post0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1266)) (2.9.0.post0)
Requirement already satisfied: python-dotenv==1.0.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1272)) (1.0.1)
Requirement already satisfied: python-fsutil==0.14.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1276)) (0.14.1)
Requirement already satisfied: python-ipware==3.0.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1280)) (3.0.0)
Requirement already satisfied: python3-openid==3.2.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1284)) (3.2.0)
Requirement already satisfied: python3-saml==1.16.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1288)) (1.16.0)
Requirement already satisfied: pytz==2024.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1293)) (2024.2)
Requirement already satisfied: pyyaml==6.0.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1299)) (6.0.2)
Requirement already satisfied: qrcode==8.0 in ./env/lib/python3.11/site-packages (from qrcode[pil]==8.0->-r /opt/inventree/src/backend/requirements.txt (line 1357)) (8.0)
Requirement already satisfied: rapidfuzz==3.10.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1363)) (3.10.1)
Requirement already satisfied: redis==5.2.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1453)) (5.2.0)
Requirement already satisfied: referencing==0.35.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1457)) (0.35.1)
Requirement already satisfied: requests==2.32.3 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1463)) (2.32.3)
Requirement already satisfied: rpds-py==0.21.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1469)) (0.21.0)
Requirement already satisfied: sentry-sdk==2.19.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1563)) (2.19.0)
Requirement already satisfied: setuptools==75.6.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1569)) (75.6.0)
Requirement already satisfied: sgmllib3k==1.0.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1575)) (1.0.0)
Requirement already satisfied: six==1.16.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1578)) (1.16.0)
Requirement already satisfied: sqlparse==0.5.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1584)) (0.5.2)
Requirement already satisfied: structlog==24.4.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1590)) (24.4.0)
Requirement already satisfied: tablib==3.5.0 in ./env/lib/python3.11/site-packages (from tablib[html,ods,xls,xlsx,yaml]==3.5.0->-r /opt/inventree/src/backend/requirements.txt (line 1594)) (3.5.0)
Requirement already satisfied: tinycss2==1.4.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1600)) (1.4.0)
Requirement already satisfied: tomli==2.1.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1607)) (2.1.0)
Requirement already satisfied: typing-extensions==4.12.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1611)) (4.12.2)
Requirement already satisfied: tzdata==2024.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1622)) (2024.2)
Requirement already satisfied: uritemplate==4.1.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1626)) (4.1.1)
Requirement already satisfied: urllib3==2.2.3 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1632)) (2.2.3)
Requirement already satisfied: wcwidth==0.2.13 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1639)) (0.2.13)
Requirement already satisfied: weasyprint==62.3 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1643)) (62.3)
Requirement already satisfied: webencodings==0.5.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1649)) (0.5.1)
Requirement already satisfied: whitenoise==6.8.2 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1657)) (6.8.2)
Requirement already satisfied: wrapt==1.17.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1661)) (1.17.0)
Requirement already satisfied: xlrd==2.0.1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1731)) (2.0.1)
Requirement already satisfied: xlwt==1.3.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1735)) (1.3.0)
Requirement already satisfied: xmlsec==1.3.14 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1739)) (1.3.14)
Requirement already satisfied: zipp==3.21.0 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1799)) (3.21.0)
Requirement already satisfied: zopfli==0.2.3.post1 in ./env/lib/python3.11/site-packages (from -r /opt/inventree/src/backend/requirements.txt (line 1803)) (0.2.3.post1)
WARNING: Ignoring invalid distribution ~etuptools (/opt/inventree/env/lib/python3.11/site-packages)
WARNING: Ignoring invalid distribution ~etuptools (/opt/inventree/env/lib/python3.11/site-packages)
Installing plugin packages from '/etc/inventree/plugins.txt'
WARNING: Ignoring invalid distribution ~etuptools (/opt/inventree/env/lib/python3.11/site-packages)
WARNING: Ignoring invalid distribution ~etuptools (/opt/inventree/env/lib/python3.11/site-packages)
WARNING: Ignoring invalid distribution ~etuptools (/opt/inventree/env/lib/python3.11/site-packages)
Python version 3.11.2 - /opt/inventree/env/bin/python3
Traceback (most recent call last):
  File "/opt/inventree/src/backend/InvenTree/manage.py", line 24, in <module>
    main()
  File "/opt/inventree/src/backend/InvenTree/manage.py", line 20, in main
    execute_from_command_line(sys.argv)
  File "/opt/inventree/env/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/opt/inventree/env/lib/python3.11/site-packages/django/core/management/__init__.py", line 416, in execute
    django.setup()
  File "/opt/inventree/env/lib/python3.11/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/opt/inventree/env/lib/python3.11/site-packages/django/apps/registry.py", line 116, in populate
    app_config.import_models()
  File "/opt/inventree/env/lib/python3.11/site-packages/django/apps/config.py", line 269, in import_models
    self.models_module = import_module(models_module_name)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/opt/inventree/src/backend/InvenTree/build/models.py", line 34, in <module>
    import InvenTree.helpers_model
  File "/opt/inventree/src/backend/InvenTree/InvenTree/helpers_model.py", line 17, in <module>
    from PIL import Image
  File "/opt/inventree/env/lib/python3.11/site-packages/PIL/Image.py", line 97, in <module>
    from . import _imaging as core
**ImportError: cannot import name '_imaging' from 'PIL' (/opt/inventree/env/lib/python3.11/site-packages/PIL/__init__.py)
ERROR: InvenTree command failed: 'python3 manage.py collectplugins'**
- Refer to the error messages in the log above for more information

**Comment by SchrodingersGat** [comment]
@tomd-tc thanks for reporting - looks to be a similar issue to what we see in https://github.com/inventree/InvenTree/issues/8495

**Comment by SchrodingersGat** [comment]
Can you please try the following: 

1. Activate the inventre python venv e.g. `/opt/inventree/env/bin/activate`
2. launch python `python`
3. List the python path (see below)

```python
import sys
print(sys.path)
```
